### ISSUE TITLE
Persona integration

### ISSUE THREAD

**Issue by cdelahousse** [SteamedPears/Code-Review#39]
_Imported issue by **[spratt](http://github.com/spratt)** from 2012-07-03T13:02:22Z_
_Originally opened as https://github.com/SteamedPears/CodeReviewClientMaster/issues/14_

---

Add Open ID integration for users who wish to log in.  I would like anonymous submissions to remain an option, so we can compete with paste bin.

P.S. Persona may be better than Open ID.

**Comment by spratt** [comment]
Peter brought up the good point of: why do we want to verify identity?  My answer was: so users can reserve names.

**Comment by spratt** [comment]
@psimonyi discuss.

**Comment by bheesham** [comment]
We should also verify identity for the not-so-public code review projects.

So, someone wants to code review a private repo on Github? Have them log in, and manage that "code review project" from some panel.

For now, I don't really see us needing it. For now, open source projects are good enough. But I think we should build with this in mind.

**Comment by psimonyi** [comment]
I like the idea of optional logging in. @cdelahousse was pointing out that a key advantage of CodeReview is the low entry barrier, because you don't have to log in, so optional is good.

Also, Mozilla Persona doesn't require an extra trip to our server the way OpenID does.

**Comment by bheesham** [comment]
Does Mozilla Persona work with different browsers?

**Comment by psimonyi** [comment]
Yes, it works in basically everything after IE7.
https://developer.mozilla.org/en-US/docs/Persona/Browser_compatibility

**Comment by spratt** [comment]
I would be happy with Persona.  Mainly what I'm looking for is the simplest implementation of letting users verify their identity if they want to.  Ideally without making them create another login, and ideally without storing any identifying information on our server.

**Comment by spratt** [comment]
I really like the ideas immediately presented on the Persona page: https://developer.mozilla.org/en-US/docs/Persona

**Comment by cdelahousse** [comment]
What about facebook integration to log in?

**Comment by spratt** [comment]
w.r.t. facebook: I wouldn't implement it myself, but I don't see a compelling reason to argue against it.

**Comment by bheesham** [comment]
I'd rather not use Facebook. Facebook is more... social. Co-workers may not necessarily want to add each other to Facebook.

**Comment by cdelahousse** [comment]
Its just another option to let people log in. I agree with what you're
saying, but fb log ins would lower the barrier of entry for a lot of
people. Stack overflow uses fob.

**Comment by bheesham** [comment]
How would it lower the barrier of entry for people?

Shawn Ellis didn't enable his Facebook up until we created a fake account.
My old roommate Eric didn't enable his Facebook until maybe near the end of the year (first year).

You can't count on people having Facebook. What you can count on, is them having an email, and a modern web browser.

**Comment by bheesham** [comment]
Seeing as how you would need an email address, and a modern web browser to even register for Facebook.

**Comment by cdelahousse** [comment]
I'm not implying that we should have it as the only log in method. Its j st
good to have it as an option for those who are comfy logging via FB. I use
that feature all the time on a ton of sites.

Some people don't use fb. Yes. They'll be catered to, also

This is not a big deal. I don't' care that much. Its just that fob
integration does not remove value but adds some for many.

**Comment by psimonyi** [comment]
This bug is now "Persona Integration". Make a new issue for Facebook or other possible identity authorities.

**Comment by spratt** [comment]
I know this issue has been resolved but it's blocking on issues with the database.  What do we need exactly?  I vaguely remember a conversation about this.  I think we need to add a users table and indicate how the user was verified.  @psimonyi let me know.

**Comment by psimonyi** [comment]
I basically have a client-side login prototype. We'd need to have a
concept of user identities on the server, plus a small server module to
verify the client's identity assertions, and presumably some changes to
the client to actually show the new things.

This is blocking on the server handling sessions and user identities,
and more or less blocking on implementing the design changes. At least,
right now I care more about some of those design changes than unblocking
this. (I might change my mind later, though.)

**Comment by spratt** [comment]
I have a redis conversion blocked waiting for another pull request to go through, but once it's in it should be easy enough to start throwing users into the database (no schema, yay!).
